scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,151 @@
+    assert os.path.exists(filename)
+    assert not csvreport_df.empty
+
+    mydict = {'test': [], 'arch' : [], 'platform' : [], 'status': [],
+              'extra_args': [], 'handler': [], 'handler_time': [],
+              'ram_size': [], 'rom_size': []}
+    assert set(mydict.keys()) == set(csvreport_df.columns)
+
+    for instance in class_testsuite.instances.values():
+        mydict["test"].append(instance.testcase.name)
+        mydict["arch"].append(instance.platform.arch)
+        mydict["platform"].append(instance.platform.name)
+        mydict["status"].append(instance.status)
+        args = " ".join(instance.testcase.extra_args) if instance.testcase.extra_args else None
+        mydict["extra_args"].append(args)
+        mydict["handler"].append(instance.platform.simulation)
+        mydict["handler_time"].append(instance.metrics.get("handler_time", None))
+        mydict["ram_size"].append(instance.metrics.get("ram_size", 0))
+        mydict["rom_size"].append(instance.metrics.get("rom_size", 0))
+
+    for i in range(len(mydict)):
+        assert set(csvreport_df[list(mydict.keys())[i]]) == set(list(mydict.values())[i])
+
+def test_xunit_report(class_testsuite, test_data, instances_fixture, tmpdir_factory):
+    """ Testing xunit_report function of Testsuite class in sanitycheck
+	Test 1: Assert sanitycheck.xml file exists after execution of xunit_report function
+	Test 2: Assert on fails, passes, skips, errors values
+	Test 3: Assert on the tree structure of sanitycheck.xml file"""
+    class_testsuite.instances = instances_fixture
+    class_testsuite.outdir = tmpdir_factory.mktemp("sanity_out_demo")
+    inst1 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_1")
+    inst2 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_2")
+    inst1.status = "failed"
+    inst2.status = "skipped"
+    filename = test_data + "sanitycheck.xml"
+    class_testsuite.xunit_report(filename)
+    assert os.path.exists(filename)
+
+    filesize = os.path.getsize(filename)
+    assert filesize != 0
+
+    fails, passes, errors, skips = 0, 0, 0, 0
+    for instance in class_testsuite.instances.values():
+        if instance.status in ["failed", "timeout"]:
+            if instance.reason in ['build_error', 'handler_crash']:
+                errors += 1
+            else:
+                fails += 1
+        elif instance.status == 'skipped':
+            skips += 1
+        else:
+            passes += 1
+
+    tree = ET.parse(filename)
+    assert int(tree.findall('testsuite')[0].attrib["skip"]) == int(skips)
+    assert int(tree.findall('testsuite')[0].attrib["failures"]) == int(fails)
+    assert int(tree.findall('testsuite')[0].attrib["errors"]) == int(errors)
+    assert int(tree.findall('testsuite')[0].attrib["tests"]) == int(passes+fails+skips+errors)
```

Comment:
why are you casting `skips, fails, errrors` and `passes + fails + skip + errors` ?

Reply:
This is the logic in sanitylib script where the report sanitycheck.xml is written in the xunit_report function. I am just asserting here that the report is written in the way it is expected by sanitylib script. See line  2990 in sanitylib.                                       

```tests="%d" % (errors + passes + fails + skips),
```